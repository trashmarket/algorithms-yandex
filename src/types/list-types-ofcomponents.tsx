import { ElementStates } from "./element-states";

export interface LetterStep {
    letter: string;
    state?: ElementStates;
    index: number
}

export interface TFibStep {
    index: number,
    fibNum: number
}