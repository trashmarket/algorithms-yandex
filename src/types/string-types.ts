import { ElementStates } from "./element-states";

export interface LetterStep {
    letter: string;
    state?: ElementStates;
    index: number
}