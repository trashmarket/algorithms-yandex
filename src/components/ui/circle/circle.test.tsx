import { render, fireEvent } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

describe("circle test", () => {
  it("circle whithout all", () => {
    const { getByTestId } = render(<Circle />);
    const circle = getByTestId("testCicrle");
    expect(circle).toMatchSnapshot();
  });

  it("circle whith letter", () => {
    const { getByTestId } = render(<Circle letter="hello" />);
    const circle = getByTestId("testCicrle");
    expect(circle).toMatchSnapshot();
  });

  it("circle whith head", () => {
    const { getByTestId } = render(<Circle head="hello head" />);
    const circle = getByTestId("testCicrle");
    expect(circle).toMatchSnapshot();
  });

  it("circle whith React element", () => {
    const { getByTestId } = render(<Circle head={<div>react Element</div>} />);
    const circle = getByTestId("testCicrle");
    expect(circle).toMatchSnapshot();
  });

  it("circle whith React element tail", () => {
    const { getByTestId } = render(<Circle tail={<div>react Element</div>} />);
    const circle = getByTestId("testCicrle");
    expect(circle).toMatchSnapshot();
  });

  it("circle whith tail", () => {
    const { getByTestId } = render(<Circle tail="tail" />);
    const circle = getByTestId("testCicrle");
    expect(circle).toMatchSnapshot();
  });

  it("circle whith index", () => {
    const { getByTestId } = render(<Circle index={1} />);
    const circle = getByTestId("testCicrle");
    expect(circle).toMatchSnapshot();
  });

  it("circle whith small props", () => {
    const { getByTestId } = render(<Circle isSmall={true} />);
    const circle = getByTestId("testCicrle");
    expect(circle).toMatchSnapshot();
  });

  it("circle whith default", () => {
    const { getAllByTestId } = render(
      <>
        <Circle state={ElementStates.Default} />
        <Circle state={ElementStates.Changing} />
        <Circle state={ElementStates.Modified} />
      </>
    );
    const circle = getAllByTestId("testCicrle");
    expect(circle).toMatchSnapshot();
  });
});
