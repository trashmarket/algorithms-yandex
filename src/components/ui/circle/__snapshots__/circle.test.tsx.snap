// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`circle test circle whith React element 1`] = `
<div
  class="content "
  data-testid="testCicrle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div>
      react Element
    </div>
  </div>
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle test circle whith React element tail 1`] = `
<div
  class="content "
  data-testid="testCicrle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div>
      react Element
    </div>
  </div>
</div>
`;

exports[`circle test circle whith default 1`] = `
Array [
  <div
    class="content "
    data-testid="testCicrle"
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>,
  <div
    class="content "
    data-testid="testCicrle"
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   changing"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>,
  <div
    class="content "
    data-testid="testCicrle"
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   modified"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>,
]
`;

exports[`circle test circle whith head 1`] = `
<div
  class="content "
  data-testid="testCicrle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head string"
  >
    hello head
  </div>
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle test circle whith index 1`] = `
<div
  class="content "
  data-testid="testCicrle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    class="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`circle test circle whith letter 1`] = `
<div
  class="content "
  data-testid="testCicrle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    >
      hello
    </p>
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle test circle whith small props 1`] = `
<div
  class="content "
  data-testid="testCicrle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle  small default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`circle test circle whith tail 1`] = `
<div
  class="content "
  data-testid="testCicrle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`circle test circle whithout all 1`] = `
<div
  class="content "
  data-testid="testCicrle"
>
  <div
    class="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    class="circle   default"
  >
    <p
      class="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    class="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    class="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
